subprojects/messaging/src/main/java/org/gradle/internal/remote/internal/inet/SocketConnection.java

```diff
@@ -190,8 +190,8 @@ public int read(byte[] dest, int offset, int max) throws IOException {
                     nread = socket.read(buffer);
                 } catch (IOException e) {
                     if (isEndOfStream(e)) {
-                        buffer.position(0);
-                        buffer.limit(0);
+                        ((Buffer)buffer).position(0);
```

Comment:
Can we add a comment somewhere in this class? Or even better create a method which converts buffer to the right type?